conda/models/environment.py

```diff
@@ -8,19 +8,91 @@
 from logging import getLogger
 from typing import TYPE_CHECKING
 
-from ..base.constants import PLATFORMS
+from ..base.constants import (
+    PLATFORMS,
+)
 from ..exceptions import CondaValueError
 
 if TYPE_CHECKING:
-    from typing import Any
-
+    from ..base.constants import (
+        ChannelPriority,
+        DepsModifier,
+        SatSolverChoice,
+        UpdateModifier,
+    )
     from .match_spec import MatchSpec
     from .records import PackageRecord
 
 
 log = getLogger(__name__)
 
 
+@dataclass
+class EnvironmentConfig:
+    """
+    **Experimental** While experimental, expect both major and minor changes across minor releases.
+
+    Data model for a conda environment config.
+    """
+
+    aggressive_update_packages: bool | None = None
+
+    channel_priority: ChannelPriority | None = None
+
+    channels: list[str] = field(default_factory=list)
+
+    channel_settings: dict[str, str] = field(default_factory=dict)
+
+    deps_modifier: DepsModifier | None = None
+
+    disallowed_packages: list[str] = field(default_factory=list)
+
+    pinned_packages: list[str] = field(default_factory=list)
+
+    repodata_fns: list[str] = field(default_factory=list)
+
+    solver: SatSolverChoice | None = None
+
+    track_features: list[str] = field(default_factory=list)
+
+    update_modifier: UpdateModifier | None = None
+
+    use_only_tar_bz2: bool | None = None
+
+    def merge(self, other):
```

Comment:
```suggestion
    @classmethod 
    def merge(cls, *other: EnvironmentConfig) -> EnvironmentConfig:
```

Don't we usually set these as `classmethod`s? At least in other APIs like `MatchSpec`, there's a precedent.

If you prefer a regular method, should we accept several `other`?